cmd/swapd/main.go

```diff
@@ -605,6 +605,14 @@ func newBackend(
 		}
 	}
 
+	extendedEC, err := extethclient.NewEthClient(ctx, env, ec, ethPrivKey)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: add configs for different eth testnets + L2 and set gas limit based on those, if not set (#153)
+	extendedEC.SetGasPrice(uint64(c.Uint(flagGasPrice)))
+	extendedEC.SetGasLimit(uint64(c.Uint(flagGasLimit)))
```

Comment:
I moved this block up a little, so the new validations would happen before the contact address is checked below.